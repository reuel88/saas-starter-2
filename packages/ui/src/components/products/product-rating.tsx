import { FC } from "react";
import { MdStar } from "react-icons/md";
import { cn } from "@repo/utilities";

type ProductRatingProps = Readonly<{
  rating: number;
}>;

export const ProductRating: FC<ProductRatingProps> = ({ rating }) => {
  const positiveRating = rating >= 0 ? rating : 0;
  const ratingRounded = Math.round(positiveRating);

  return (
    <div className="flex">
      {new Array(5).fill("").map((_, index) => (
        <MdStar
          key={index}
          className={cn(
            "h-5 w-5",
            index < ratingRounded ? "text-hue-color" : "text-muted",
          )}
        />
      ))}
      <span className="text-muted-foreground">({rating})</span>
    </div>
  );
};
