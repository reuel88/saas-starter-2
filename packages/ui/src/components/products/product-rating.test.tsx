import { render, screen } from "@testing-library/react";
import { ProductRating } from "./product-rating";

describe("ProductRating", () => {
  it("should render 4 stars when rating", () => {
    // Arrange

    // Act
    const { baseElement } = render(<ProductRating rating={4} />);

    // Assert
    expect(screen.getByText("(4)")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  it("should render 0 stars when negative rating", () => {
    // Arrange

    // Act
    const { baseElement } = render(<ProductRating rating={-4} />);

    // Assert
    expect(screen.getByText("(-4)")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
