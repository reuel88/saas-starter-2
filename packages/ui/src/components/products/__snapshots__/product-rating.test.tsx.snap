// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductRating > should render 0 stars when negative rating 1`] = `
<body>
  <div>
    <div
      class="flex"
    >
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <span
        class="text-muted-foreground"
      >
        (
        -4
        )
      </span>
    </div>
  </div>
</body>
`;

exports[`ProductRating > should render 4 stars when rating 1`] = `
<body>
  <div>
    <div
      class="flex"
    >
      <svg
        class="h-5 w-5 text-hue-color"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-hue-color"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-hue-color"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-hue-color"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <svg
        class="h-5 w-5 text-muted"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <span
        class="text-muted-foreground"
      >
        (
        4
        )
      </span>
    </div>
  </div>
</body>
`;
